deepchecks/analytics/anonymous_telemetry.py

```diff
@@ -14,29 +14,117 @@
+
+    return 'native'
+
+
+try:
+    RUNTIME = _identify_runtime()
+except Exception:  # pylint: disable=broad-except
+    RUNTIME = 'unknown'
+
 
+def get_environment_details():
+    """Get environment details."""
+    return {
+        'python_version': platform.python_version(),
+        'os': platform.system(),
+        'deepchecks_version': deepchecks.__version__,
+        'runtime': RUNTIME,
+    }
 
-def send_anonymous_import_event():
-    """Send an anonymous import event to PostHog."""
-    if not ANALYTICS_DISABLED:
+
+def send_anonymous_event(event_name, event_data=None):
+    """Send an anonymous event ."""
+    if ANALYTICS_ENABLED:
         try:
-            if os.path.exists(os.path.join(MODULE_DIR, '.user_id')):
-                with open(os.path.join(MODULE_DIR, '.user_id'), 'r', encoding='utf8') as f:
-                    user_id = f.read()
-            else:
-                user_id = str(uuid.uuid4())
-                with open(os.path.join(MODULE_DIR, '.user_id'), 'w', encoding='utf8') as f:
-                    f.write(user_id)
-
-            conn = http.client.HTTPSConnection('api.deepchecks.com', timeout=3)
-            conn.request('GET', f'/metrics?version={deepchecks.__version__}&uuid={user_id}')
+            machine_id = get_telemetry_config()['machine_id']
+            params = {
+                'event_name': event_name,
+                'event_data': event_data,
+                'machine_id': machine_id,
+            }
+            params.update(get_environment_details())
+
+            conn = http.client.HTTPConnection('localhost:80', timeout=3)
+            conn.request('POST', '/events', body=json.dumps(params))
             _ = conn.getresponse()
         except Exception:  # pylint: disable=broad-except
             pass
+
+
+def send_anonymous_run_event(run_instance):
+    """Send an anonymous check run event."""
+    from deepchecks import BaseSuite, BaseCheck  # pylint: disable=import-outside-toplevel
+    if isinstance(run_instance, BaseCheck):
+        event_name = 'run-check'
+    elif isinstance(run_instance, BaseSuite):
+        event_name = 'run-suite'
+    else:
+        event_name = 'run-unknown'
+
+    try:
+        mod = run_instance.__module__
+
+        def get_custom_check_kind(clazz):
+            """Get first deepchecks ancestor."""
+            mro = clazz.__class__.__mro__
+            for cls in mro:
+                if cls.__module__.startswith('deepchecks'):
+                    return cls.p__module__.split('.')[1]
+            return 'unknown'
+
+        if mod.startswith('deepchecks'):
+            try:
+                name = run_instance.name()
+            except Exception:  # pylint: disable=broad-except
+                name = run_instance.__class__.__name__
+            event_data = {
+                'name': name,
+                'kind': mod.split('.')[1],
+            }
+        else:
+            event_data = {
+                'name': 'custom_check',
```

Comment:
a. it might be custom suite
b. we don't log the checks inside suite?

Reply:
a - you're right. will modify
b - no